docs/reference/contrib/modeladmin/migrating_to_snippets.md

```diff
@@ -0,0 +1,141 @@
+
+### Using ModelAdmin to manage Page models
+
+ModelAdmin allows the registration of Page models, but the main use case is to create custom page listing views. The create and edit views are not supported by ModelAdmin, as there are page-specific operations in those views that are best handled by Wagtail's page views. For this reason, registering a Page model as a snippet is not supported.
+
+In a future release, Wagtail will introduce a new "treeless" listing view for pages, as outlined in [RFC 082: Treeless page listings](https://github.com/wagtail/rfcs/pull/82) and the [Universal Listings discussion](https://github.com/wagtail/wagtail/discussions/10446). This feature will allow for custom page listing views and will be the recommended way to achieve this use case.
+
+### Customisation of index view table rows and columns
+
+ModelAdmin has a number of APIs that allow customisation of the index view table rows and columns. Meanwhile, Wagtail has an internal generic tables UI framework that is used throughout the admin, including snippets. This table framework will become the standard way to build table elements in index views within the admin. As a result, the following APIs are not supported in snippets:
+
+- `ModelAdmin.get_extra_attrs_for_row`
+
+  This can be achieved by creating a custom `wagtail.admin.ui.tables.Table` subclass and using it as the `IndexView.table_class`.
+
+- `ModelAdmin.get_extra_class_names_for_field_col`
+
+  This can be achieved using a custom `wagtail.admin.ui.tables.Column` instance in `SnippetViewSet.list_display`.
+- `ModelAdmin.list_display_add_buttons`
+
+  By default, the first column specified in `list_display` is the one that contains the buttons. Using custom `wagtail.admin.ui.tables.Column` instances in `SnippetViewSet.list_display` allows you to specify a different column.
+
+- Attributes for `wagtail.contrib.modeladmin.mixins.ThumbnailMixin`
+
+  This mixin is used to show a thumbnail in the index view. A similar functionality can be achieved using a custom `wagtail.admin.ui.tables.Column` instance in `SnippetViewSet.list_display`. Hence, the following attributes are not supported:
+
+    - `ModelAdmin.thumb_image_field_name`
+    - `ModelAdmin.thumb_image_width`
+    - `ModelAdmin.thumb_classname`
+    - `ModelAdmin.thumb_col_header_text`
+    - `ModelAdmin.thumb_default`
+
+### Custom CSS and JS
+
+ModelAdmin supports inserting custom extra CSS and JS files into the admin via the following attributes on the ModelAdmin class:
+
+- `ModelAdmin.index_view_extra_css`
+- `ModelAdmin.index_view_extra_js`
+- `ModelAdmin.form_view_extra_css`
+- `ModelAdmin.form_view_extra_js`
+- `ModelAdmin.inspect_view_extra_css`
+- `ModelAdmin.inspect_view_extra_js`
+
+This is not supported in snippets, but custom CSS and JS files can be inserted by overriding the respective view's template.
```

Comment:
Is it worth calling out the hooks also here?

Reply:
Yep, I think so. Updated.